src/sys/node/node-lazy-require.ts

```diff
@@ -1,43 +1,61 @@
 import type * as d from '../../declarations';
 import { buildError } from '@utils';
 import { NodeResolveModule } from './node-resolve-module';
-import semiver from 'semiver';
 import fs from 'graceful-fs';
 import path from 'path';
+import satisfies from 'semver/functions/satisfies';
+import major from 'semver/functions/major';
 
+/**
+ * The version range that we support for a given package
+ * [0] is the lower end, while [1] is the higher end.
+ *
+ * These strings should be standard semver strings.
+ */
+type NodeVersionRange = [string, string];
+
+/**
+ * A manifest for lazily-loaded dependencies, mapping dependency names
+ * to version ranges.
+ */
+type LazyDependencies = Record<string, NodeVersionRange>;
+
+/**
+ * Lazy requirer for Node, with functionality for specifying version ranges
+ * and returning diagnostic errors if requirements aren't met.
+ */
 export class NodeLazyRequire implements d.LazyRequire {
   private ensured = new Set<string>();
 
-  constructor(
-    private nodeResolveModule: NodeResolveModule,
-    private lazyDependencies: { [dep: string]: [string, string] }
-  ) {}
+  constructor(private nodeResolveModule: NodeResolveModule, private lazyDependencies: LazyDependencies) {}
 
   async ensure(fromDir: string, ensureModuleIds: string[]) {
```

Comment:
Minor, can we JSDoc this function while we're in here?

Reply:
addressed in https://github.com/ionic-team/stencil/pull/3346/commits/e452f2a0b6eb3becbb97fc8b44c7366d665e1b52

Reply:
Ahh sorry if I wasn't clear here - meant on the `ensure()` method since we were getting it under test. The ctor JSDoc looks good though! Let's keep that 🙂

Reply:
no worries! just added that in https://github.com/ionic-team/stencil/pull/3346/commits/8c9032212c025eb1d5bb506b0e0098a74c30f3da